bentoml/deployment/aws_lambda/__init__.py

```diff
@@ -0,0 +1,425 @@
+                    s3_bucket_name=lambda_s3_bucket,
+                    py_runtime=python_runtime,
+                    memory_size=aws_config.memory_size,
+                    timeout=aws_config.timeout,
+                )
+                validate_lambda_template(template_file_path)
+                logger.debug('Lambda project directory: {}'.format(lambda_project_dir))
+                logger.debug('initializing lambda project')
+                init_sam_project(
+                    lambda_project_dir,
+                    bento_path,
+                    deployment_pb.name,
+                    deployment_spec.bento_name,
+                    api_names,
+                )
+                logger.info('Packaging lambda project')
+                lambda_package(
+                    lambda_project_dir, lambda_s3_bucket, deployment_path_prefix
+                )
+                logger.info('Deploying lambda project')
+                try:
+                    lambda_deploy(
+                        lambda_project_dir,
+                        stack_name=deployment_pb.namespace + '-' + deployment_pb.name,
+                    )
+                except Exception as e:
+                    _cleanup_s3_bucket(lambda_s3_bucket, aws_config.region)
+                    return ApplyDeploymentResponse(status=Status.INTERNAL(str(e)))
+
+            logger.info('Finish deployed lambda project, fetching latest status')
+            res_deployment_pb = Deployment(state=DeploymentState())
+            res_deployment_pb.CopyFrom(deployment_pb)
+            state = self.describe(res_deployment_pb, yatai_service).state
+            res_deployment_pb.state.CopyFrom(state)
+            return ApplyDeploymentResponse(
+                status=Status.OK(), deployment=res_deployment_pb
+            )
+
+        except BentoMLException as error:
+            return ApplyDeploymentResponse(status=exception_to_return_status(error))
+
+    def delete(self, deployment_pb, yatai_service):
+        try:
+            logger.debug('Deleting AWS Lambda deployment')
+            state = self.describe(deployment_pb, yatai_service).state
+            if state.state != DeploymentState.RUNNING:
+                message = (
+                    'Failed to delete, no active deployment {name}. '
+                    'The current state is {state}'.format(
+                        name=deployment_pb.name,
+                        state=DeploymentState.State.Name(state.state),
+                    )
+                )
+                return DeleteDeploymentResponse(status=Status.ABORTED(message))
+
+            deployment_spec = deployment_pb.spec
+            aws_config = deployment_spec.aws_lambda_operator_config
+
+            cf_client = boto3.client('cloudformation', aws_config.region)
+            stack_name = deployment_pb.namespace + '-' + deployment_pb.name
+            cloud_formation_stack_result = cf_client.describe_stacks(
```

Comment:
use the returned value from self.describe above, you already have the s3 bucket information in the returned info_json

Reply:
which can save you one cf_client.describe_stacks call to AWS